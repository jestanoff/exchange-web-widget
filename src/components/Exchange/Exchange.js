import React, { useEffect } from 'react';
import root from 'window-or-global';
import { func, shape, string } from 'prop-types';
import Button from '../Forms/Button';
import ExchangeRate from '../ExchangeRate';
import Pocket from '../Pocket';
import SwapPocketsButton from '../SwapPocketsButton';
import * as currencies from '../../constants/currencies';
import styles from './exchange.css';
const FETCH_INTERVAL = 10000;
const symbols = ['GBP', 'EUR', 'USD'];
const Exchange = ({ fromPocket, getCurrencyExchangeRates, rate, rates, toPocket }) => {
  useEffect(() => {
    getCurrencyExchangeRates('GBP', symbols);
    const id = root.setInterval(() => getCurrencyExchangeRates('GBP', symbols), FETCH_INTERVAL);
    return () => root.clearInterval(id);
  }, []);

  return (
    <main className={styles.container}>
      <form name="exchange" action="/">
        <Pocket amount="109.54" currency={fromPocket.currency} type="from" />
        <section className={styles.exchangeRateContainer}>
          <SwapPocketsButton />
          <ExchangeRate from={fromPocket.currency} rate={rates.EUR} to={toPocket.currency} />
        </section>
        <Pocket currency={toPocket.currency} type="to" />
        <Button />
      </form>
    </main>
  );
};

Exchange.propTypes = {
  fromPocket: shape({ currency: string }),
  getCurrencyExchangeRates: func.isRequired,
  rate: string,
  toPocket: shape({ currency: string }),
};

Exchange.defaultProps = {
  fromPocket: { currency: currencies.USD.code },
  rate: '1.2039',
  rates: {},
  toPocket: { currency: currencies.CHF.code },
};

export default Exchange;
