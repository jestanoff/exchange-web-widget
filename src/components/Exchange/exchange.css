.container {
  padding: 0;
}

.exchangeRateContainer {
  display: flex;
  justify-content: space-between;
  margin-top: -18px;
}
