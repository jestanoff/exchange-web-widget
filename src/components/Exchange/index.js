import { connect } from 'react-redux';
import getCurrencyExchangeRates from 'actions/getCurrencyExchangeRates';
import Exchange from './Exchange';

const mapPropsToState = state => ({
  rates: state && state.fxRates && state.fxRates.rates,
});

const mapDispatchToState = {
  getCurrencyExchangeRates,
};

export default connect(mapPropsToState, mapDispatchToState)(Exchange);
