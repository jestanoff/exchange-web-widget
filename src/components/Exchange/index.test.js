import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Exchange from './';


describe('Exchange component', () => {
  test('should render proper markup', () => {
    const { container } = render(<Exchange />);
  
    expect(container).toMatchSnapshot();
  })
});
