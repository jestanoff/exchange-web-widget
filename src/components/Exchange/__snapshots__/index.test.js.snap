// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exchange component should render proper markup 1`] = `
<div>
  <main
    class="app-container"
  >
    <form
      action="/"
      name="exchange"
    >
      <section
        class="pocket"
      >
        <span>
          EUR
        </span>
        <input
          name="from"
          type="text"
        />
      </section>
      <span>
        Swap
      </span>
      <section
        class="exchange-rate"
      >
        â‚¬
         1 = 
        fr
         
        1.2039
      </section>
      <section
        class="pocket"
      >
        <span>
          CHF
        </span>
        <input
          name="to"
          type="text"
        />
      </section>
      <button
        type="submit"
      >
        Exchange
      </button>
    </form>
  </main>
</div>
`;
